__tests__/hooks/headless/useChatMessages.test.tsx

```diff
@@ -0,0 +1,162 @@
+import { MedplumClient } from "@medplum/core";
+import { Communication, Patient } from "@medplum/fhirtypes";
+import { MockClient } from "@medplum/mock";
+import { MedplumProvider } from "@medplum/react-hooks";
+import { act, renderHook, waitFor } from "@testing-library/react-native";
+
+import { useChatMessages } from "@/hooks/headless/useChatMessages";
+
+const mockPatient: Patient = {
+  resourceType: "Patient",
+  id: "test-patient",
+  name: [{ given: ["John"], family: "Doe" }],
+};
+
+const mockThread: Communication = {
+  resourceType: "Communication",
+  id: "test-thread",
+  status: "completed",
+  sent: "2024-01-01T12:00:00Z",
+};
+
+const mockMessage1: Communication = {
+  resourceType: "Communication",
+  id: "msg-1",
+  status: "completed",
+  sent: "2024-01-01T12:01:00Z",
+  sender: {
+    reference: "Patient/test-patient",
+    display: "John Doe",
+  },
+  payload: [{ contentString: "Hello" }],
+  partOf: [{ reference: "Communication/test-thread" }],
+};
```

Comment:
One approach is to use `createReference` so that you don't have to make too many adjustments when the referenced object changes

```suggestion
const mockMessage1: Communication = {
  resourceType: "Communication",
  id: "msg-1",
  status: "completed",
  sent: "2024-01-01T12:01:00Z",
  sender: createReference(mockPatient),
  payload: [{ contentString: "Hello" }],
  partOf: [createReference(mockThread)],
};
```